util/util.go

```diff
@@ -878,28 +886,38 @@ func TrimFilesystem(volumeName string, encryptedDevice bool) error {
 		deviceDir = EncryptedDeviceDirectory
 	}
 
-	mountOutput, err := nsExec.Execute("bash", []string{"-c", fmt.Sprintf("cat /proc/mounts | grep %s%s | awk '{print $2}'", deviceDir, volumeName)})
-	if err != nil {
-		return errors.Wrapf(err, "cannot find volume %v mount info on host", volumeName)
-	}
+	var validMountpoint string
+	lines := strings.Split(content, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			continue
+		}
 
-	mountList := strings.Split(strings.TrimSpace(mountOutput), "\n")
+		device := fields[0]
+		if !strings.HasPrefix(device, deviceDir+volumeName) {
+			continue
+		}
 
-	var mountpoint string
-	for _, m := range mountList {
-		_, err = nsExec.Execute("stat", []string{m})
+		mountPoint := fields[1]
+		_, err = lhns.GetFileInfo(mountPoint)
 		if err == nil {
-			mountpoint = m
+			validMountpoint = mountPoint
 			break
 		}
+	}
 
-		logrus.WithError(err).Warnf("Failed to get volume %v mount point %v info", volumeName, m)
+	if validMountpoint == "" {
+		return fmt.Errorf("failed to find valid mountpoint")
 	}
-	if mountpoint == "" {
-		return fmt.Errorf("cannot find a valid mount point for volume %v", volumeName)
+
+	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
+	nsexec, err := lhns.NewNamespaceExecutor(lhns.GetDefaultProcessName(), lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		return err
 	}
 
-	_, err = nsExec.Execute("fstrim", []string{mountpoint})
+	_, err = nsexec.Execute(lhtypes.BinaryFstrim, []string{validMountpoint}, lhtypes.ExecuteDefaultTimeout)
```

Comment:
Suggest having a unit test for this mount point parsing from the content of mounts

Reply:
We can have unit tests to cover the validation of the mount point instead of just the content. This will give us a bit more coverage.

Reply:
Sounds good.